### PR TITLE
AVL Tree

### PR DESCRIPTION
**Pull request by FilipeFalcaoBatista**
AVL Tree data structure and its test.

### CODE REVIEW

File: src/data_structures/avl_tree.js [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+    current = node;
+  }
+  return this._findMax(node.right, current);
+};
+
+/**
+ * Finds the node with minimum value in the whole tree.
+ */
+AVLTree.prototype.findMin = function() {
+  return this._findMin(this.root);
```
Comment (zzz6519003) [inline src/data_structures/avl_tree.js]: _findMin   and findMin    what's the best about this XD  additional method thing

**Comment by FilipeFalcaoBatista** [comment]
Nothing special. _findMin is the real method, a generic one. findMin just calls _findMin with the root, so walks the hole tree to find the minimum. Just created a new method to simplify a very repeating task. Same idea for _findMax and findMax.

File: src/data_structures/avl_tree.js (hunk 1 above)
Comment (FilipeFalcaoBatista) [inline src/data_structures/avl_tree.js]: Nothing special. _findMin is the real method, a generic one. findMin just calls _findMin with the root, so walks the hole tree to find the minimum. Just created a new method to simplify a very repeating task. Same idea for _findMax and findMax.

**Comment by felipernb** [comment]
Thanks for the contribution

**Pull request merged**